iree/turbine/kernel/ops/wave_ops.py

```diff
@@ -1152,6 +1181,34 @@ def align_index(self, constraints: list["Constraint"]) -> None:
         if is_shared_mem_access(self):
             self.index = align_index_vars(self.index, constraints)
 
+    def has_identity_mapping(self) -> bool:
```

Comment:
Since these are identical, can we add them to the CustomOp or you could create a new base class for Read and Write?

Reply:
They are not totally identical, I use `mapping.input_shape` for `read` and `mapping.output_shape` for `write`.

Reply:
I think this can still be refactored but we can save that for a future PR.